code/datums/storage/storage.dm

```diff
@@ -1117,10 +1055,55 @@ GLOBAL_LIST_EMPTY(cached_storage_typecaches)
 
 	modeswitch_action = null
 
+/// Updates views of all objects in storage and stretches UI to appropriate size
+/datum/storage/proc/orient_storage()
+	var/adjusted_contents = length(real_location.contents)
+	var/list/datum/numbered_display/numbered_contents
+	if(numerical_stacking)
+		numbered_contents = process_numerical_display()
+		adjusted_contents = length(numbered_contents)
+
+	//if the ammount of contents reaches some multiplier of the final column (and its not the last slot), let the player view an additional row
+	var/additional_row = (!(adjusted_contents % screen_max_columns) && adjusted_contents < max_slots)
+
+	var/columns = clamp(max_slots, 1, screen_max_columns)
+	var/rows = clamp(CEILING(adjusted_contents / columns, 1) + additional_row, 1, screen_max_rows)
+
+	for (var/ui_user in storage_interfaces)
+		storage_interfaces[ui_user].update_position(screen_start_x, screen_pixel_x, screen_start_y, screen_pixel_y, columns, rows)
+
+	var/current_x = screen_start_x
+	var/current_y = screen_start_y
+	var/turf/our_turf = get_turf(real_location)
+
+	var/list/obj/storage_contents = list()
+	if (islist(numbered_contents))
+		for(var/content_type in numbered_contents)
+			var/datum/numbered_display/numberdisplay = numbered_contents[content_type]
+			storage_contents[numberdisplay.sample_object] = MAPTEXT("<font color='white'>[(numberdisplay.number > 1)? "[numberdisplay.number]" : ""]</font>")
+	else
+		for(var/obj/item as anything in real_location)
+			storage_contents[item] = ""
+
+	for(var/obj/item as anything in storage_contents)
+		item.mouse_opacity = MOUSE_OPACITY_OPAQUE
+		item.screen_loc = "[current_x]:[screen_pixel_x],[current_y]:[screen_pixel_y]"
+		item.maptext = storage_contents[item]
+		item.plane = ABOVE_HUD_PLANE
+		SET_PLANE(item, ABOVE_HUD_PLANE, our_turf)
```

Comment:
missed from previous review? only need the second plane set, first one is deprecated